src/nl/surf/eduhub_rio_mapper/utils/keystore.clj

```diff
@@ -56,8 +61,35 @@
   {:post [(some? (:certificate %))]}
   (let [ks (keystore keystore-path keystore-pass)]
     {:keystore        ks
+     :keystore-alias  keystore-alias
      :trust-store     (keystore-resource "truststore.jks" "")
      :keystore-pass   keystore-pass
      :trust-store-pass ""
      :private-key     (get-key ks keystore-alias keystore-pass)
      :certificate     (get-certificate ks keystore-alias keystore-pass)}))
+
+(defn validate-keystore
+  "Count the number of entries in a keystore file.
+   Returns a map with :count and :aliases keys."
+  [keystore-path password]
+  (let [h
+          (try
+            (let [keystore (KeyStore/getInstance "JKS")
+                  fis (FileInputStream. keystore-path)]
+              (try
+                (.load keystore fis (when password (.toCharArray password)))
+                (let [aliases (Collections/list (.aliases keystore))
+                      count (.size keystore)]
+                  {:count count
+                  :aliases (vec aliases)
+                  :success true})
+                (finally
+                  (.close fis))))
```

Comment:
Gebruik `with-open` ipv `(try .. (finally (.close ..)))`

Of liever nog `keystore-resource` in `nl.surf.eduhub-rio-mapper.utils.keystore` -- of is er een reden om die niet te gebruiken?